app/services/google_parse_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+class GoogleParseService
+  def initialize(html:)
+    @html = html
+    @document = Nokogiri::HTML(html)
+  end
+
+  def result
+    {
+      ads_top_count: ads_top_count,
+      ads_page_count: ads_page_count,
+      non_ads_count: non_ads_count,
+      total_links_count: total_links_count,
+      all_links: all_links
+    }
+  end
+
+  private
+
+  attr_reader :html, :document
+
+  # Number of AdWords advertisers in the top position.
+  def ads_top_count
+    document.css('#tads > div').count
+  end
```

Comment:
How about doing caching here so it does not need to count every time when this method get called?
```suggestion
  def ads_top_count
    @ads_top_count ||= document.css('#tads > div').count
  end
```

Reply:
Sorry for this late reply!
Fixed in [`5f004ca` (#47)](https://github.com/suho/google-search-ruby/pull/47/commits/5f004cae025caf7b86ce910239bd5478850fb86a)